packages/ag-charts-website/src/features/api-documentation/components/ApiReference.tsx

```diff
@@ -1,284 +1,261 @@
 import Code from '@components/Code';
-import { LinkIcon } from '@components/link-icon/LinkIcon';
+import { Icon } from '@components/icon/Icon';
 import { useToggle } from '@utils/hooks/useToggle';
 import classnames from 'classnames';
-import { createContext, useContext } from 'react';
+import type { AllHTMLAttributes } from 'react';
+import { createContext, useContext, useEffect } from 'react';
 import Markdown from 'react-markdown';
 
-import type { ApiReferenceType } from '../api-reference-types';
-import styles from './ApiDocumentation.module.scss';
-import { PropertyName } from './PropertyName';
-import { ToggleDetails } from './ToggleDetails';
+import type { ApiReferenceNode, ApiReferenceType, MemberNode, TypeAliasNode } from '../api-reference-types';
+import { formatTypeToCode, getMemberType, isInterfaceHidden } from '../utils/apiReferenceHelpers';
+import { scrollIntoViewById, useLocation } from '../utils/navigation';
+import styles from './ApiReference.module.scss';
+import { PropertyTitle, PropertyType } from './Properies';
 
 export const ApiReferenceContext = createContext<ApiReferenceType | null>(null);
+export const ApiReferenceConfigContext = createContext<ApiReferenceConfig>({});
 
-const hiddenInterfaces = [
-    'CssColor',
-    'FontStyle',
-    'FontWeight',
-    'FontSize',
-    'FontFamily',
-    'Opacity',
-    'PixelSize',
-    'Ratio',
-];
-
-interface ApiReferenceOptions {
-    id: string;
-    reference: ApiReferenceType;
+interface ApiReferenceConfig {
+    prioritise?: string[];
     include?: string[];
     exclude?: string[];
     hideHeader?: boolean;
-    displayFirst?: string[];
+    hideRequired?: boolean;
+    specialTypes?: Record<string, string>;
```

Comment:
Maybe the value of the `Record` could be more specific, to indicate possible values